Tests/ArgumentParserUnitTests/ParsableArgumentsValidationTests.swift

```diff
@@ -193,4 +193,127 @@ final class ParsableArgumentsValidationTests: XCTestCase {
     }
     try PositionalArgumentsValidator.validate(K.self)
   }
+
+  private struct DifferentNames: ParsableArguments {
+    @Option()
+    var foo: String
+
+    @Option()
+    var bar: String
+  }
+
+  struct TwoOfTheSameName: ParsableCommand {
+    @Option()
+    var foo: String
+
+    @Option(name: .customLong("foo"))
+    var notActuallyFoo: String
+  }
+
+  private struct MultipleUniquenessViolations: ParsableArguments {
+    @Option()
+    var foo: String
+
+    @Option(name: .customLong("foo"))
+    var notActuallyFoo: String
+
+    @Option()
+    var bar: String
+
+    @Flag(name: .customLong("bar"))
+    var notBar: Bool
+
+    @Option()
+    var help: String
+  }
+
+  private struct MultipleNamesPerArgument: ParsableCommand {
+    @Flag(name: [.customShort("v"), .customLong("very-chatty")])
+    var verbose: Bool
+
+    enum Versimilitude: String, ExpressibleByArgument {
+      case yes
+      case some
+      case none
+    }
+
+    @Option(name: .customShort("v"))
+    var versimilitude: Versimilitude
+  }
+
+  private struct FourDuplicateNames: ParsableArguments {
+    @Option()
+    var foo: String
+
+    @Option(name: .customLong("foo"))
+    var notActuallyFoo: String
+
+    @Flag(name: .customLong("foo"))
+    var stillNotFoo: Bool
+
+    enum Numbers: Int, ExpressibleByArgument {
+      case one = 1
+      case two
+      case three
+    }
+
+    @Option(name: .customLong("foo"))
+    var alsoNotFoo: Numbers
+  }
+
```

Comment:
There are no tests for `@Flag`s using `CaseIterable` enums. The code does the right thing but having a test case might be useful in the future.

```swift
enum ExampleEnum: String, ExpressibleByArgument, CaseIterable {
    case first
    case second
    case other
    case forth
    case fith
}

struct Example: ParsableCommand {
  @Flag(name: .short, default: .first) //this should fail
  var enumFlag: ExampleEnum

  @Flag(name: .long, default: .first) //this should pass
  var enumFlag2: ExampleEnum
}
```

Reply:
Thank you very much--I didn't know about this potential source of conflict. Tests added for the long and short name cases.